src/transport_ofi.c

```diff
@@ -1408,7 +1408,7 @@ int shmem_transport_init(void)
 
     if (shmem_internal_params.OFI_PROVIDER_provided)
         shmem_transport_ofi_info.prov_name = shmem_internal_params.OFI_PROVIDER;
-    if (shmem_internal_params.OFI_USE_PROVIDER_provided)
+    else if (shmem_internal_params.OFI_USE_PROVIDER_provided)
         shmem_transport_ofi_info.prov_name = shmem_internal_params.OFI_USE_PROVIDER;
     else
         shmem_transport_ofi_info.prov_name = NULL;
```

Comment:
This `else` case was resetting `shmem_transport_ofi_info.prov_name` after it was set from `OFI_PROVIDER` above.  🤦‍♂️